docs/data_types/tensor/tensor.md

```diff
@@ -0,0 +1,118 @@
+
+## Tensor typed validation
+
+You don't need to instantiate directly the  [`NdArray`][docarray.typing.tensor.NdArray] , [`TorchTensor`][docarray.typing.tensor.TorchTensor], [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] directly by yourself. They are only use as type hint field fot the [`BaseDoc`][docarray.base_doc.doc.BaseDoc] validation.
+[`BaseDoc`][docarray.base_doc.doc.BaseDoc] will cast the native tensor type into the respective DocArray tensor type.
+
+Let's take an example 
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+import numpy as np
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray
+
+
+doc = MyDoc(tensor=np.zeros(100))
+
+assert isinstance(doc.tensor, NdArray)  # True
+``` 
+here you see that the `doc.tensor` is a `NdArray`. 
+
+```python
+assert isinstance(doc.tensor, np.ndarray)  # True as well
+``` 
+
+But since it inherits from `np.ndarray` you can use it as a numpy array.
+
+## Type coercion from numpy to pytorch 
+
+!!! note
+    If you pass a numpy array to a pytorch tensor field, the numpy array will be converted to a pytorch tensor. 
+
+
+example:
+
+
+```python
+from docarray import BaseDoc
+from docarray.typing import TorchTensor
+import numpy as np
+
+
+class MyTensorsDoc(BaseDoc):
+    tensor1: TorchTensor[512]
+    tensor2: TorchTensor[512]
+
+
+rand_series_f64 = np.random.rand(512).astype('float64')
+rand_series_f32 = np.random.rand(512).astype('float32')
+
+doc = MyTensorsDoc(tensor1=rand_series_f64, tensor2=rand_series_f32)
+doc.summary()
+```
+
+```
+📄 MyTensorsDoc : 84877dd ...
+╭──────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────╮
+│ Attribute            │ Value                                                                                    │
+├──────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────┤
+│ tensor1: TorchTensor │ TorchTensor of shape (512,), dtype: torch.float64                                        │
+│ tensor2: TorchTensor │ TorchTensor of shape (512,), dtype: torch.float32                                        │
+╰──────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────╯
+```
+
+
+But this is not the case if you pass a numpy array to a tensorflow tensor field. 
+
+!!! warning 
+    Tensor field type coercion is only supported from numpy to pytorch, not the other way around.
+
+
+## DocVec with AnyTensor
+
+DocVec can be used with a BaseDoc which has a field of [`AnyTensor`][docarray.typing.tensor.AnyTensor] or a Union of tensor type. 
+
+ But to do so DocVec needs to know the tensor type of the tensor field beforehand to create the right column.
+ 
+You can precise these parameters with the `tensor_type` parameter of the [`DocVec`][docarray.vectorizer.doc_vec.DocVec] constructor.
```

Comment:
No it is only about Union. Otherwise it already know what is the expected tensor types.

This has been changed very early in the DocVec design I think